### ISSUE TITLE
Issue: 24-hour Weather Forecast has invalid start-end dates

### ISSUE THREAD

**Issue by teddyywj** [datagovsg/datagovsg-datasets#802]
### Please provide the name of the dataset or describe the dataset that you're requesting:
24-hour Weather Forecast 

### Do you know which agency provides/can provide this data?
*( Refer to this link for a full list of agencies: https://data.gov.sg/organization )*
NA

### If there's an issue with a dataset, please provide a description of the problem
For 24-hour weather forecast that involves forecasting into the next year, start-end dates are invalid. 

Example: 
[Weather Forecast @ 2019-12-31 09:00](https://api.data.gov.sg/v1/environment/24-hour-weather-forecast?date_time=2019-12-31T09%3A00%3A00) End time is invalid.

`{
                    "regions": {
                        "central": "Windy",
                        "east": "Windy",
                        "north": "Windy",
                        "south": "Windy",
                        "west": "Windy"
                    },
                    "time": {
                        "end": "2019-01-01T06:00:00+08:00",
                        "start": "2019-12-31T18:00:00+08:00"
                    }
                }`

[Weather Forecast @ 2020-01-01 09:00](https://api.data.gov.sg/v1/environment/24-hour-weather-forecast?date_time=2020-01-01T09%3A00%3A00) Start time is invalid. 

`{
                    "regions": {
                        "central": "Windy",
                        "east": "Windy",
                        "north": "Windy",
                        "south": "Windy",
                        "west": "Windy"
                    },
                    "time": {
                        "end": "2020-01-01T06:00:00+08:00",
                        "start": "2020-12-31T18:00:00+08:00"
                    }
                },`

 

### Please provide the fields expected in this dataset:
NA

### How frequently would you need this dataset to be updated?
NA

### Can you give a short explanation on how you're planning to use this dataset?
NA

**Comment by lennardl** [comment]
@teddyywj This is meant to retrieve a 24 hour forecast and not over a one year range

**Comment by teddyywj** [comment]
@lennardl  
Yes, I understand.  Error comes when the 24 hour forecast crosses over to next year (Prediction done on 31st Dec 2019 09:00 to forecast the next 24 hours). 

[Weather forecast @ 2019-12-31 09:00](https://api.data.gov.sg/v1/environment/24-hour-weather-forecast?date_time=2019-12-31T09%3A00%3A00)

The year for the end-time is invalid.  

"valid_period":{
   "start":"**2019-12-31**T06:00:00+08:00",
   "end":"**2019-01-01**T06:00:00+08:00"
}

This happens consistently when retrieving the next 24 hour forecast @ the end of the year (31st Dec) and start of the year (1st Jan).
